### DISCUSSION TITLE
Cannot access transmission from outside LAN

### DISCUSSION THREAD

**Discussion by abescalamis** [haugene/docker-transmission-openvpn#2890] [Q&A] [answered]
I used to have this working on my Synology NAS, but moved it to my Ubuntu 24.10 - Server, everything works except accessing the web GUI from outside my network, only works on LAN. (and yes the port on the router is open).

version: '3.3'
services:
    transmission-openvpn:
        container_name: Transmission
        cap_add:
            - NET_ADMIN
        volumes:
            - '/media/Downloads/:/data/downloads'
            - '/home/test/Transmission/config/:/config'
        environment:
            - OPENVPN_USERNAME=xxxxxxxxxxxxxxxxxxxxxxxx
            - OPENVPN_PASSWORD=xxxxxxxxxxxxxxxxxxxxxxxx
            - OPENVPN_PROVIDER=EXPRESSVPN
            - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
            - LOCAL_NETWORK=192.168.1.0/24
            - GLOBAL_APPLY_PERMISSIONS=true
            - TRANSMISSION_HOME=/config/transmission-home
            - TRANSMISSION_RPC_PORT=9091
            - TRANSMISSION_RPC_USERNAME=xxxxxxxxxxx
            - TRANSMISSION_RPC_PASSWORD=xxxxxxxxxxx
            - TRANSMISSION_DOWNLOAD_DIR=/data/downloads/completed
            - TRANSMISSION_INCOMPLETE_DIR=/data/downloads/incomplete
            - TRANSMISSION_WATCH_DIR=/data/downloads/watch
            - CREATE_TUN_DEVICE=true
            - ENABLE_UFW=true
            - UFW_ALLOW_GW_NET=true
            - UFW_EXTRA_PORTS=9910,23561,443
            - UFW_DISABLE_IPTABLES_REJECT=true
            - PUID=1000
            - PGID=1000
            - TZ=America/Los_Angeles
            - PEER_DNS=true
            - PEER_DNS_PIN_ROUTES=true
            - DROP_DEFAULT_ROUTE=true
            - WEBPROXY_ENABLE=false
            - WEBPROXY_USERNAME=xxxxxxxx
            - WEBPROXY_PASSWORD=xxxxxxxx
            - LOG_TO_STDOUT=false
            - HEALTH_CHECK_HOST=google.com
            - SELFHEAL=false
            - OPENVPN_CONFIG=xxxxxxxxxxxxx
            - TRANSMISSION_RPC_AUTHENTICATION_REQUIRED=true
            - TRANSMISSION_WEB_UI=transmissionic
        logging:
            driver: json-file
            options:
                max-size: 10m
        ports:
            - '9091:9091'
        restart: always
        image: haugene/transmission-openvpn

**Comment by pkishino** [comment]
That is expected.. you really shouldn’t open it up to external..
But if you really do then you need to update your local _network to allowlist the ip range of your external range or setup a reverse proxy

**Reply by abescalamis** [reply]
My local network is the traditional - 192.168.1.1 (router) with the IP Pool Ending 192.168.1.254, that is why I used - LOCAL_NETWORK=192.168.1.0/24.

How do I  allow list the IP range?

**Reply by pkishino** [reply, marked as answer]
Don’t.. setup either a reverse proxy.. or better a Tailscale/wiregate mode that you connect to from external securely which then gives you access to all your lan as if you were inside

**Reply by gajahduduk** [reply]
> My local network is the traditional - 192.168.1.1 (router) with the IP Pool Ending 192.168.1.254, that is why I used - LOCAL_NETWORK=192.168.1.0/24.
> 
> How do I allow list the IP range?

did you figure this out?  setting up a reverse proxy is probably too complicated for me, but [this unfinished since 2020 documentation](https://haugene.github.io/docker-transmission-openvpn/vpn-networking/) suggests you can do it just by setting that local_network variable.  Which I also did but it still doesn't work.  Is it necessary to also set up port forwarding on the router to the RPC port?  I had previously tried that and it worked, but it apparently did not need any authentication, so the whole things was exposed!  (yes I had set the transmission_rpc_user and _rpc_pass variables too).

**Reply by abescalamis** [reply]
Yes, I was able to solve it, with the information Pkishino provided, I bought a domain name "example.com", then I created a subdomain and put it on nginx, you also need to make sure your router has port forwarding for NGINX.

If you need help let me know.

There is many YouTube videos that show how to do the same thing (search: nginx + emby or any other program) they

**Reply by gajahduduk** [reply]
> If you need help let me know.
Thanks. I barely know where to start. Getting this haugene container up and running was very difficult.

edit: I managed to get a reverse proxy setup directly within my Synology NAS!  Apparently it has nginx built in. So much easier than configuring another container. [Excellent video tutorial here](https://www.youtube.com/watch?v=iWvCN2j7xjo) and the [written version](https://www.wundertech.net/synology-reverse-proxy-setup-config/) by same author here. I found that the DDNS service provided by my router (asus) did not work, as it appears not to support subdomains.  however, the synology DDNS service worked.
